agent/agent.go

```diff
@@ -2854,14 +2854,28 @@ func (a *Agent) deletePid() error {
 // definitions on disk, and load them into the local agent.
 func (a *Agent) loadServices(conf *config.RuntimeConfig) error {
 	// Register the services from config
+	var trimmedServices []*structs.ServiceDefinition
+	a.logger.Printf("[DEBUG] agent: The initial services are: %s", conf.Services)
+	// Go though the services and either return on first error (IgnoreBadCheckFiles == false) or continue processing
+	// other checks but leave the bad ones out from any further processing
 	for _, service := range conf.Services {
-		ns := service.NodeService()
-		chkTypes, err := service.CheckTypes()
-		if err != nil {
+		_, err := service.CheckTypes()
+		if err != nil && a.config.IgnoreBadCheckFiles == false {
 			return fmt.Errorf("Failed to validate checks for service %q: %v", service.Name, err)
+		} else {
+			if err == nil {
```

Comment:
I'm a little confused on the approach here. Could you elaborate why we are explicitly doing "else, if err = nil"? My understanding is that instead of an else/if else we could just run the code as normally?

Reply:
Hi @s-christoff - for all of the comments the common comment from my side is that I tried to maintain defaults the way consul has. So the new flag works if set to true explicitly. That way it doesn't break compatibility with existing deployments.

Given that inside `if err != nil && a.config.IgnoreBadCheckFiles == false` we're returning, I could skip its corresponding else but I have to have if-else inside, because in one case I append to services adn in the other I simply log that I am skipping that service.

Now, I could invert it to check for
```
if err != nil {
   a.logger.Printf("[WARN] agent: Going to skip service %s ", service)
  } else {
   trimmedServices = append(trimmedServices, service)
}```

I think I started with the sequence of the defaults and deviations from the defaults in my mind and so it looks that way.

Would that address this comment?

Reply:
That would!